Feature: Native Block Editor Interface

fabiankaegy opened the pull request:
> [!WARNING]
> This PR is based on top of #95

### Description of the Change

This PR adds a WordPress Data API store and Block Editor UI for managing Content Connect relationships directly within the WordPress Block Editor. 

**WordPress Data API Store (`wp-content-connect`):**

- Redux-based store for managing relationship state within the Block Editor
- Selectors: `getRelationships()`, `getRelatedEntities()`, `getDirtyEntityIds()`
- Actions: `setRelationships()`, `setRelatedEntities()`, `updateRelatedEntities()`, `markPostAsDirty()`
- Automatic persistence of relationship changes when posts are saved
- Resolvers for lazy-loading relationship data via REST API

**React Hooks:**

- `useRelationships()` - Hook for accessing and loading relationships for a post
- `useRelatedEntities()` - Hook for accessing, loading, and updating related entities for a specific relationship

**Block Editor UI:**

- `RelationshipsPanel` - Plugin that automatically adds relationship panels to the Document Settings sidebar
- `RelationshipManager` - Component that integrates with `@10up/block-components` ContentPicker for managing related content
- Panels are automatically shown for relationships with `enable_ui: true`
- Supports both post-to-post and post-to-user relationships
- Respects relationship configuration (sortable, max_items, post_type filtering)

**Key Features:**

- Automatic integration with existing REST API endpoints (from #95)
- Lazy-loading of relationship data for optimal performance
- Dirty state tracking to only persist changes when necessary
- Seamless integration with WordPress post save lifecycle
- TypeScript support with full type definitions
- Integration with `@10up/block-components` ContentPicker component

This implementation enables developers to easily build custom blocks and interfaces that interact with Content Connect relationships using standard WordPress Data API patterns.

Closes #45

### How to test the Change

1. **Setup:**
   - Ensure you have a test environment with relationships configured (e.g., using the demo plugin: https://github.com/fabiankaegy/wp-content-connect-example-data)
   - Ensure relationships have `enable_ui: true` set

2. **Block Editor UI:**
   - Open a post in the Block Editor that has relationships configured
   - Navigate to the Document Settings sidebar
   - Verify that relationship panels appear for each relationship with `enable_ui: true`
   - Test adding, removing, and reordering related entities using the ContentPicker interface
   - Save the post and verify relationships are persisted correctly

3. **Data Store API (Browser Console):**

```javascript
// Get relationships for a post
wp.data.select('wp-content-connect').getRelationships(postId)

// Get related entities for a specific relationship
wp.data.select('wp-content-connect').getRelatedEntities(postId, {
  rel_key: 'relationship_key',
  rel_type: 'post-to-post'
})

// Update related entities
wp.data.dispatch('wp-content-connect').updateRelatedEntities(
  postId,
  'relationship_key',
  'post-to-post',
  [entityId1, entityId2]
)
```

4. **React Hooks (in custom blocks/components):**

```typescript
import { useRelationships, useRelatedEntities } from '@wordpress/content-connect';

// In your component
const [hasResolved, relationships] = useRelationships(postId);
const [hasLoaded, entities, updateEntities] = useRelatedEntities(postId, {
  rel_key: 'relationship_key',
  rel_type: 'post-to-post'
});
```

### Changelog Entry

- Added - WordPress Data API store (`wp-content-connect`) for managing Content Connect relationships within the Block Editor.
- Added - React hooks `useRelationships()` and `useRelatedEntities()` for accessing and managing relationships in custom blocks.
- Added - Block Editor UI integration with automatic relationship panels in the Document Settings sidebar.

### Credits

Props @fabiankaegy @s3rgiosan

### Checklist

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added [Critical Flows, Test Cases, and/or End-to-End Tests](https://10up.github.io/Open-Source-Best-Practices/testing/) to cover my change.
- [x] All new and existing tests pass.

fabiankaegy committed:
fix update node version to 20

fabiankaegy committed:
fix update node version to 20

fabiankaegy committed:
replace view app with modern toolkit based build process

fabiankaegy committed:
replace view app with modern toolkit based build process

fabiankaegy committed:
update installed packages

fabiankaegy committed:
update installed packages

fabiankaegy committed:
add initial scaffolding for store

fabiankaegy committed:
add initial scaffolding for store

fabiankaegy committed:
fix comment out enqueued scripts from MetaBox

fabiankaegy committed:
fix comment out enqueued scripts from MetaBox

fabiankaegy committed:
add initial working version of the store

fabiankaegy committed:
add initial working version of the store

fabiankaegy committed:
clean up store

fabiankaegy committed:
clean up store

fabiankaegy committed:
add hooks for interfacing with data store

fabiankaegy committed:
add hooks for interfacing with data store

fabiankaegy committed:
harden types

fabiankaegy committed:
harden types

fabiankaegy committed:
fix how data is passed to server

fabiankaegy committed:
fix how data is passed to server

fabiankaegy committed:
handle undefined post ID’s

fabiankaegy committed:
handle undefined post ID’s

fabiankaegy committed:
install additional dependencies

fabiankaegy committed:
install additional dependencies

fabiankaegy committed:
add initial version of UI

fabiankaegy committed:
add initial version of UI

s3rgiosan committed:
Move PluginDocumentSettingsPanel to RelationshipManager component for Classic Editor usage

s3rgiosan committed:
Move PluginDocumentSettingsPanel to RelationshipManager component for Classic Editor usage

s3rgiosan committed:
Replace FormTokenField with 10up Content Picker

s3rgiosan committed:
Replace FormTokenField with 10up Content Picker

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Remove console.log

s3rgiosan committed:
Remove console.log

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Rename RelatedPosts to RelatedEntities to support post-to-user relationships

s3rgiosan committed:
Rename RelatedPosts to RelatedEntities to support post-to-user relationships

s3rgiosan committed:
Rename file

s3rgiosan committed:
Rename file

s3rgiosan committed:
Pass undefined on contentTypes for post-to-user relationships

s3rgiosan committed:
Pass undefined on contentTypes for post-to-user relationships

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Remove test code 🤦‍♂️

s3rgiosan committed:
Remove test code 🤦‍♂️

s3rgiosan commented:
Hi @fabiankaegy I went ahead and updated the UI to use the new ContentPicker, along with some fixes and improvements for post-to-user relationship support and bi-directional panels in post-to-post relationships. Ready for your review when you can.

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan commented:
I've created a `try/block-editor-ui` all-in-one branch to make it easier to test the features in both the block and classic editors, as well as the Query Loop integration.

ocean90 reviewed (commented):
This is looking pretty good already! Thanks for the try/block-editor-ui branch. Two things I noticed while doing some basic testing:
* If `sortable` is enabled sorting is a bit laggy. Sometimes the item positions are not changed.I think this happens because the sorting is directly saved via a POST request and directly followed by a GET request. There seems to be a race-condition where the second runs to early? Should saving maybe delayed until the post is saved?
* The second one is related to `enable_ui` which is simply ignored in the block editor, meaning the UI is always there.

s3rgiosan commented:
> This is looking pretty good already! Thanks for the try/block-editor-ui branch. Two things I noticed while doing some basic testing:
> 
> * If `sortable` is enabled sorting is a bit laggy. Sometimes the item positions are not changed.I think this happens because the sorting is directly saved via a POST request and directly followed by a GET request. There seems to be a race-condition where the second runs to early? Should saving maybe delayed until the post is saved?
> * The second one is related to `enable_ui` which is simply ignored in the block editor, meaning the UI is always there.

Thanks for testing this @ocean90. I'll review the issues you pointed out ASAP.

s3rgiosan committed:
Respect "enable_ui" flag when rendering the relationship panel

s3rgiosan committed:
Respect "enable_ui" flag when rendering the relationship panel

s3rgiosan commented:
> The second one is related to enable_ui which is simply ignored in the block editor, meaning the UI is always there.

@ocean90 this was resolved in the latest commit.

s3rgiosan committed:
Add await for updateRelatedEntities

s3rgiosan committed:
Add await for updateRelatedEntities

s3rgiosan commented:
> If sortable is enabled sorting is a bit laggy. Sometimes the item positions are not changed.I think this happens because the sorting is directly saved via a POST request and directly followed by a GET request. There seems to be a race-condition where the second runs to early? Should saving maybe delayed until the post is saved?

Hi @ocean90,

I haven't been able to replicate this same issue on my end, but I did notice that was missing an `await` when calling updateRelatedEntities. I added the `await` and it should help avoid timing issues.

Can you try again when you have a chance to see if the problem still persists?

If so, it would be really helpful to get a bit more context about your setup:
- For the post type, how many relationships are defined?
- How many related entities are there in each relationship (especially the one creating the problem), roughly speaking?

Once again, thank you!

s3rgiosan commented:
@fabiankaegy Is there a way to improve this experience? Maybe by using an internal state and syncing changes to the REST API afterward?


https://github.com/user-attachments/assets/d3f9e84e-2271-4284-a0d3-ded4ef398d41

fabiankaegy commented:
@s3rgiosan yeah we really should improve that. I though that the way we handle the rest API communication in the custom Data store would have already taken care of this but apparently not. 

I will try and see if I can find some time at the end of this week for this. Feel free to explore either adjusting the logic in the datastore to not persist on change directly but rather only queueing up the changes in the datastore as local changes that are in memory only and then only persist the relationships when we actually save the post 🤔 

Alternatively, we could look at using something like [`useOptimistic`](https://react.dev/reference/react/useOptimistic) in the actual component

s3rgiosan commented:
> @s3rgiosan yeah we really should improve that. I though that the way we handle the rest API communication in the custom Data store would have already taken care of this but apparently not.
> 
> I will try and see if I can find some time at the end of this week for this. Feel free to explore either adjusting the logic in the datastore to not persist on change directly but rather only queueing up the changes in the datastore as local changes that are in memory only and then only persist the relationships when we actually save the post 🤔
> 
> Alternatively, we could look at using something like [`useOptimistic`](https://react.dev/reference/react/useOptimistic) in the actual component

I'll explore some options and take a look into `useOptimistic`. Thanks!

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Use id and remove ID

s3rgiosan committed:
Use id and remove ID

s3rgiosan committed:
Update internal state while syncing to the server in the background

s3rgiosan committed:
Update internal state while syncing to the server in the background

s3rgiosan committed:
Add docblocks

s3rgiosan committed:
Add docblocks

s3rgiosan committed:
Extract key generation logic

s3rgiosan committed:
Extract key generation logic

s3rgiosan committed:
Remove redundant state update call

s3rgiosan committed:
Remove redundant state update call

s3rgiosan committed:
Memoize selector

s3rgiosan committed:
Memoize selector

s3rgiosan committed:
Fix adding new relationships

s3rgiosan committed:
Fix adding new relationships

s3rgiosan committed:
Update useRelatedEntities hook

s3rgiosan committed:
Update useRelatedEntities hook

s3rgiosan commented:
@fabiankaegy latest update on the sorting experience:


https://github.com/user-attachments/assets/f4255736-e90a-494a-a651-aea74fce2de8

ocean90 commented:
> Hi @ocean90,
> Can you try again when you have a chance to see if the problem still persists?

I just tested the latest changes and the problem doesn't exist anymore. Good job!

Something to consider: The content picker doesn't work well with longer titles. The component uses `<Truncate>` with no option (like a tooltip) to view the full title anymore. If you have very similar titles, you will no longer be able to distinguish between them.

s3rgiosan commented:
> Something to consider: The content picker doesn't work well with longer titles. The component uses `<Truncate>` with no option (like a tooltip) to view the full title anymore. If you have very similar titles, you will no longer be able to distinguish between them.

Thanks for the suggestion, @ocean90! I'm actually working on a custom picker item for Content Connect here: https://github.com/10up/wp-content-connect/pull/99. I'll keep that in mind when I get back to it.

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Improve ContentPicker query filtering for relationship context

s3rgiosan committed:
Improve ContentPicker query filtering for relationship context

s3rgiosan commented on assets/js/components/relationship-manager.tsx:
```diff
@@ -0,0 +1,50 @@
+			queryFilter={(query) => {
+				if (relationship?.rel_key) {
+					return addQueryArgs(query, {
+						content_connect: relationship.rel_key
```
I added this search parameter mainly to give context to the editor-side search when it reaches the `rest_post_search_query` hook in PHP. One of my clients uses ElasticPress, and I needed a way to disable EP for these specific queries. This parameter can also be used to filter search results based on client-specific requirements. It might also be worth including the current post ID.

s3rgiosan committed:
Update MetaBox.php

Remove commented code

s3rgiosan committed:
Update MetaBox.php

Remove commented code

barryceelen committed:
Fix typo

barryceelen committed:
Fix typo

barryceelen committed:
Add support for max_items

The JS that renders the content picker already added support for a new
max_items key on the relationship args. This update adds the required
PHP update to support that key.

barryceelen committed:
Add support for max_items

The JS that renders the content picker already added support for a new
max_items key on the relationship args. This update adds the required
PHP update to support that key.

s3rgiosan committed:
Merge pull request #112 from 10up/feature/max-items

Feature/max items

s3rgiosan committed:
Merge pull request #112 from 10up/feature/max-items

Feature/max items

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	includes/API/V2/Route/Relationships.php

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	includes/API/V2/Route/Relationships.php

s3rgiosan committed:
Update to 100

s3rgiosan committed:
Update to 100

s3rgiosan committed:
Update 10up Block components

s3rgiosan committed:
Update 10up Block components

s3rgiosan committed:
Add npm scripts

s3rgiosan committed:
Add npm scripts

s3rgiosan committed:
Merge branch 'feature/helpers' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/helpers' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	includes/Plugin.php

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	includes/Plugin.php

s3rgiosan committed:
Only reflect relationship changes after post being saved

s3rgiosan committed:
Only reflect relationship changes after post being saved

fabiankaegy commented on assets/js/store/index.ts:
```diff
@@ -0,0 +1,272 @@
+	 */
+	markPostAsDirty(postId: number): MarkPostAsDirtyAction {
+		// Trigger the block editor to mark the post as dirty.
+		(dispatch('core/editor') as any).editPost({
```
Instead of calling dispatch on the string `core/editor` we should import the `store` from the `@wordpress/editor` package. Doing so will allow typescript to understand what's available in the store :) 

```ts
import { store as editorStore } from '@wordpress/editor';

dispatch(editorStore).editPost({});
```
```

s3rgiosan replied:
This was addressed. Thanks @fabiankaegy

s3rgiosan committed:
Address PR comments

s3rgiosan committed:
Address PR comments

s3rgiosan committed:
Update dependencies

s3rgiosan committed:
Update dependencies

ocean90 commented:
Noticed an issue with the display of relationships which can't be edited by the user: https://github.com/10up/block-components/issues/390

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	.nvmrc
#	composer.lock
#	includes/UI/MetaBox.php
#	package-lock.json
#	package.json

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	.nvmrc
#	composer.lock
#	includes/UI/MetaBox.php
#	package-lock.json
#	package.json

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Update 10up block components dependecy

s3rgiosan committed:
Update 10up block components dependecy

s3rgiosan committed:
Remove URL from UI

s3rgiosan committed:
Remove URL from UI

s3rgiosan commented:
@rickalee PR is ready for review. The `1.7.0-beta.1` tag on `release/1.7.0` should help with testing in your projects. Let me know if you find any issues.

cc @jeffpaul @fabiankaegy

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	package-lock.json

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	package-lock.json

s3rgiosan committed:
Restoring URL

Moving implementation to a hook based logic in `feature/picked-relationship-component`

s3rgiosan committed:
Restoring URL

Moving implementation to a hook based logic in `feature/picked-relationship-component`

s3rgiosan committed:
Add compiled assets

s3rgiosan committed:
Add compiled assets

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	package-lock.json

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	package-lock.json

s3rgiosan committed:
Ignore dist dir on spellcheck

s3rgiosan committed:
Ignore dist dir on spellcheck

s3rgiosan committed:
Fix typo definition

s3rgiosan committed:
Fix typo definition

s3rgiosan committed:
Fix file format

s3rgiosan committed:
Fix file format

s3rgiosan committed:
Update dependencies

s3rgiosan committed:
Update dependencies

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	includes/Plugin.php
#	package-lock.json
#	package.json

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	includes/Plugin.php
#	package-lock.json
#	package.json

s3rgiosan committed:
Fix previous commit

s3rgiosan committed:
Fix previous commit

s3rgiosan committed:
no message

s3rgiosan committed:
no message

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	tests/unit/bootstrap.php

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	tests/unit/bootstrap.php

s3rgiosan committed:
Add missing key to map

s3rgiosan committed:
Add missing key to map

s3rgiosan committed:
Serialize options to avoid unnecessary re-renders when parent re-renders

s3rgiosan committed:
Serialize options to avoid unnecessary re-renders when parent re-renders

s3rgiosan committed:
Add try-catch to api fetch requests

s3rgiosan committed:
Add try-catch to api fetch requests

s3rgiosan committed:
Filter non number ids

s3rgiosan committed:
Filter non number ids

s3rgiosan committed:
Memoize options to avoid unnecessary re-renders when parent re-renders

s3rgiosan committed:
Memoize options to avoid unnecessary re-renders when parent re-renders

s3rgiosan committed:
Update built assets

s3rgiosan committed:
Update built assets

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	includes/Plugin.php
#	tests/unit/ContentConnectUnitTestCase.php

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	includes/Plugin.php
#	tests/unit/ContentConnectUnitTestCase.php

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Fetch all pages to ensure all related entities are loaded

s3rgiosan committed:
Fetch all pages to ensure all related entities are loaded

s3rgiosan committed:
Add fixed height for content types that have many related entities

s3rgiosan committed:
Add fixed height for content types that have many related entities

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

s3rgiosan committed:
Rename test php command

s3rgiosan committed:
Rename test php command

s3rgiosan committed:
Merge branch 'feature/data-api-backbone' of github.com:10up/wp-content-connect into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/rest-api-endpoint' into feature/data-api-backbone

# Conflicts:
#	.gitignore
#	includes/API/V2/Post/Route/AbstractPostRoute.php
#	includes/API/V2/Route/Relationships.php
#	includes/Helpers.php
#	package-lock.json
#	package.json
#	phpunit.unit.xml
#	tests/php/API/V1/SearchTest.php
#	tests/php/PluginTest.php
#	tests/php/Tables/BaseTableTest.php
#	tests/php/UI/MetaBoxTest.php
#	tests/php/UI/PostToPostTest.php
#	tests/php/UI/PostToUserTest.php
#	tests/unit/ContentConnectUnitTestCase.php
#	tests/unit/UI/MetaBoxTest.php
#	tests/unit/bootstrap.php

s3rgiosan committed:
Merge branch 'feature/helpers' into feature/data-api-backbone

s3rgiosan committed:
Merge branch 'feature/helpers' into feature/data-api-backbone

# Conflicts:
#	includes/UI/MetaBox.php

s3rgiosan committed:
Address copilot review on data api store

s3rgiosan committed:
Re-add legacy ID key for backwards compatibility

s3rgiosan committed:
Sync package-lock version to 2.0.0 and restore build deps

s3rgiosan committed:
Bump dependency-review workflow actions to v5

s3rgiosan committed:
Fix dependency-review failures

s3rgiosan committed:
Merge branch 'develop' into feature/data-api-backbone

s3rgiosan committed:
Sync package-lock to resolve npm ci out-of-sync failure

s3rgiosan committed:
Fix output release zip into release/ dir for artifact upload

s3rgiosan committed:
Cover BlockEditor enqueue and relationship max_items

s3rgiosan committed:
Add jest coverage for the block editor data store

s3rgiosan committed:
Add wp-env-seeded Playwright E2E suite

s3rgiosan committed:
Sync package-lock.json so npm ci resolves dependencies
